### PR TITLE
feat: [#188489104] anytime action dropdown categories

### PR DESCRIPTION
**Pull request by adidner**
## Description

We wanted to add categories to anytime action dropdown

### Ticket

This pull request resolves [#188489104](https://www.pivotaltracker.com/story/show/188489104).

### Approach

We took some time to refactor the how the CMS is structured to better represent these new categories which is a large part of the file changes. 

### Steps to Test

Form as a business, Go through the show funding card too,

See the anytime action drop down, 

see the categories

you can also go to the dev modifyBusiness URL and then add a license as expired to see that the anytime action liense reinstatements still work. 

you can also go to the anytime action tasks themselves to make sure they still work.

## Code author checklist

- [X] I have rebased this branch from the latest main branch
- [X] I have performed a self-review of my code
- [X] I have created and/or updated relevant documentation on the engineering documentation website
- [X] I have not used any relative imports
- [X] I have pruned any instances of unused code
- [X] I have not added any markdown to labels, titles and button text in config
- [X] If I added/updated any values in `userData` (including `profileData`, `formationData` etc), then I added a new migration file
- [X] I have checked for and removed instances of unused config from CMS
- [X] If I added any new collections to the CMS config, then I updated the search tool and `cmsCollections.ts` (see CMS Additions in Engineering Reference/FAQ on the engineering documentation site)
- [X] I have updated relevant `.env` values in both `.env-template` and in Bitwarden

### CODE REVIEW

File: web/src/components/dashboard/AnytimeActionDropdown.tsx [hunk 1]
```diff
@@ -14,49 +14,126 @@ import { useRouter } from "next/router";
...
+  const getApplicableAnytimeActions = (): AnytimeActionWithTypeAndCategory[] => {
     const anytimeActionLinkWithType = props.anytimeActionLinks
       .filter((action) => findMatch(action))
       .map((action) => {
-        return { ...action, type: "link" };
+        return {
+          ...action,
+          type: "link",
+          category: Config.dashboardAnytimeActionDefaults.anytimeActionDropdownCategoryAdmin,
+        };
```
Comment (cbrunefearless) [inline web/src/components/dashboard/AnytimeActionDropdown.tsx:74]: Nice work here

File: web/src/lib/async-content-fetchers/fetchAnytimeActionByFilename.test.ts [hunk 2]
```diff
@@ -1,12 +1,24 @@
...
 describe("fetchAnytimeActionByFileName", () => {
-  it("returns a record when a valid id is supplied", async () => {
+  it("returns a record when a valid admin id is supplied", async () => {
     const record = await fetchAnytimeActionByFilename("njstart-registration");
     expect(record.filename).toEqual("njstart-registration");
     expect(record.contentMd.length).toBeGreaterThan(0);
   });
 
+  it("returns a record when a valid licenses id is supplied", async () => {
+    const record = await fetchAnytimeActionByFilename("carnival-ride-supplemental-modification");
```
Comment (cbrunefearless) [inline web/src/lib/async-content-fetchers/fetchAnytimeActionByFilename.test.ts:11]: should it be singular license? sorry nit pickey

File: content/src/anytime-action-tasks-licenses/electrical-contracting-business-permit-update.md [hunk 3]
```diff
@@ -26,33 +26,34 @@ summaryDescriptionMd: >-
...
-* Notarized affidavit
-* Applicable fees paid
-* Name change information, if applicable
-* Business address and license number if business permit has ever been issued under a different license
-* Name and address of stakeholders with vested ownership, if the business is not a corporation
+- `EIN|ein`
+- Electrical contractor’s name and license number
+- Expiration date of last renewal
+- Business address
+- `Surety bond|surety-bond` for $1,000
```
Comment (cbrunefearless) [inline content/src/anytime-action-tasks-licenses/electrical-contracting-business-permit-update.md:36]: Just me ranting, I hate how our linter does this

**Review by cbrunefearless — APPROVED** [review]
just me ranting and a small change.. otherwise good work!

File: web/src/lib/async-content-fetchers/fetchAnytimeActionByFilename.ts [hunk 4]
```diff
@@ -2,23 +2,38 @@ import { AnytimeActionTask } from "@/lib/types/types";
 import { convertAnytimeActionTaskMd } from "@/lib/utils/markdownReader";
 
 export const fetchAnytimeActionByFilename = async (id: string): Promise<AnytimeActionTask> => {
+  // This is bad and should 100% be refactored to not exist and is as a high priority refactor
```
[the remark this answers is not in the document]
Comment (adidner) [reply web/src/lib/async-content-fetchers/fetchAnytimeActionByFilename.ts:5]: https://www.pivotaltracker.com/story/show/188619340

Yes I added this, ticket above

**Pull request merged**